src/limbo/kernel/squared_exp_ard.hpp

```diff
@@ -71,9 +71,15 @@ namespace limbo {
             double operator()(const Eigen::VectorXd& x1, const Eigen::VectorXd& x2) const
             {
                 assert(x1.size() == _ell.size());
-                Eigen::MatrixXd K = (_A * _A.transpose());
-                K.diagonal() += (Eigen::MatrixXd)(_ell.array().inverse().square());
-                double z = ((x1 - x2).transpose() * K * (x1 - x2)).norm();
+                double z;
+                if (Params::kernel_squared_exp_ard::k() > 0) {
+                    Eigen::MatrixXd K = (_A * _A.transpose());
+                    K.diagonal() += (Eigen::MatrixXd)(_ell.array().inverse().square());
+                    z = ((x1 - x2).transpose() * K * (x1 - x2)).norm();
+                }
+                else {
+                    z = (x1 - x2).cwiseQuotient(_ell).squaredNorm();
+                }
                 return _sf2 * std::exp(-0.5 * z);
```

Comment:
> Oo ok... Operation on empty matrices are not optimized during compilation? I thought, in the worst case, it would caused only a handful of operations, but not such delay

I wasn't expecting that delay also. But apparently that was the issue. Maybe the zero size in one dimension could have caused some issues.. Be sure to re-run them to be sure..